main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"go_service/src"
+	"go_service/src/adapters"
+	"go_service/src/api"
+	"go_service/src/common"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+	"net/http"
+	"os"
+)
+
+func setupHandlers(cr common.CurrencyReader, sa api.SubscriberGateway) *http.ServeMux {
+	sm := http.NewServeMux()
+	sm.HandleFunc("/", api.GetCurrencyHandler(cr))
+	sm.HandleFunc("/subscribe", api.GetSubscribersHandler(sa))
```

Comment:
These function names do not give the user an understanding of what kind of http method they expect. Instead, I would change them to: GetCurrentCurrency, PostSubscribeUser. This way they are self-documented - they say what method they expect.

Reply:
Yes, you are right. I named this functions because they are factories, but these names actually confuse the reader